zerver/views/messages.py

```diff
@@ -603,11 +603,12 @@ def get_old_messages_backend(request, user_profile,
 
 @has_request_variables
 def update_message_flags(request, user_profile,
-                      messages=REQ('messages', validator=check_list(check_int)),
-                      operation=REQ('op'), flag=REQ('flag'),
-                      all=REQ('all', validator=check_bool, default=False)):
+                         messages=REQ('messages', validator=check_list(check_int)),
+                         operation=REQ('op'), flag=REQ('flag'),
+                         all=REQ('all', validator=check_bool, default=False),
+                         stream_name=REQ('stream_name', default=None)):
     request._log_data["extra"] = "[%s %s]" % (operation, flag)
-    do_update_message_flags(user_profile, operation, flag, messages, all)
+    do_update_message_flags(user_profile, operation, flag, messages, all, stream_name)
```

Comment:
This needs a check and some error handling for the case that user isn't subscribed to a stream by the name `stream_name`.  Probably the right way to do it is to move the `get_stream` call up to this function, and pass the stream object to `do_update_messages_flags`.  And if `get_stream` fails, return a nice `json_error` message.

Also, can you add a backend test for the case of sending an invalid stream name?